crates/network-scanner/Cargo.toml

```diff
@@ -17,10 +17,19 @@ network-scanner-proto ={ path = "../network-scanner-proto" }
 parking_lot = "0.12.1"
 socket2 = "0.5.5"
 thiserror = "1.0.57"
-tokio = { version = "1.36.0", features = ["rt", "sync", "time"] }
+tokio = { version = "1.36.0", features = ["rt", "sync", "time", "fs"] }
 tracing = "0.1.40"
 typed-builder = "0.18.1"
 
+[target.'cfg(target_os = "windows")'.dependencies]
+ipconfig = "0.3.2"
+
+[target.'cfg(target_os = "linux")'.dependencies]
+futures = "0.3.30"
```

Comment:
suggestion: You can use `futures-util` instead. This is a smaller dependency.

Reply:
updated